src/components/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,208 @@
+import { useParams, useSearchParams } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
+import { Person } from '../../types';
+import '../../App.scss';
+import { PersonLink } from '../PersonLink';
+import { SearchLink } from '../SearchLink';
+
+const SORT_TITLES = ['name', 'sex', 'born', 'died'];
+
+type Props = {
+  people: Person[],
+};
+
+export const PeopleTable: React.FC<Props> = ({ people }) => {
+  const [searchParams] = useSearchParams();
+  const { slug } = useParams();
+  const [visiblePeople, setVisiblePeople] = useState(people);
+
+  const selectedPerson = slug;
+
+  const query = searchParams.get('query')?.toLowerCase();
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
+  useEffect(() => {
+    let filteredPeople = sex
+      ? people.filter(person => person.sex === sex)
+      : [...people];
+
+    if (query) {
+      filteredPeople = filteredPeople.filter((obj) => {
+        const queryLow = query.toLowerCase();
+
+        return (
+          (obj.name && obj.name.toLowerCase().includes(queryLow))
+          || (obj.motherName && obj.motherName.toLowerCase().includes(queryLow))
+          || (obj.fatherName && obj.fatherName.toLowerCase().includes(queryLow))
+        );
+      });
+    }
+
+    if (centuries.length > 0) {
+      filteredPeople = filteredPeople.filter((person) => {
+        const personCentury = Math.ceil(person.born / 100);
+
+        if (centuries.includes(personCentury.toString())) {
+          return person;
+        }
+
+        return null;
+      });
+    }
+
+    if (sort === 'born' || sort === 'died') {
+      filteredPeople = filteredPeople.sort((a, b) => a[sort] - b[sort]);
+    }
+
+    if (sort === 'name' || sort === 'sex') {
+      filteredPeople = (
+        filteredPeople.sort((a, b) => a[sort].localeCompare(b[sort]))
+      );
+    }
```

Comment:
- it's better to use switch case statement instead of multiple if blocks 
- please also move 'born', 'died', 'name' and 'sex' to the enum as it will make a code cleaner